### PR TITLE
Add pypi installer support

### PR DESCRIPTION
**Pull request by tfcollins**

### CODE REVIEW

**Comment by tfcollins** [comment]
For appveyor CI I might suggest we not use bdist_wininst anymore and just distribute wheels.

File: bindings/python/setup.py.cmakein [hunk 1]
```diff
@@ -13,11 +13,56 @@
...
-		url='https://github.com/analogdevicesinc/libiio',
-		py_modules=['iio'],
-		)
+import setuptools
+from setuptools.command.install import install
+
+with open("../../README.md", "r") as fh:
+    long_description = fh.read()
+
+
```
Comment (commodo) [inline bindings/python/setup.py.cmakein]: nitpick: extra line;

File: bindings/python/setup.py.cmakein [hunk 2]
```diff
@@ -13,11 +13,56 @@
...
+        try:
+            _lib = _cdll(find_library(_iiolib), use_errno=True, use_last_error=True)
+        except:
+            msg = "The libiio library could not be found.\n\
+            libiio needs to be installed first before the python bindings.\n\
+            The latest release can be found on GitHub:\n\
+            https://github.com/analogdevicesinc/libiio/releases"
+            raise Exception(msg)
+
+
```
Comment (commodo) [inline bindings/python/setup.py.cmakein]: nitpick: extra line;

File: bindings/python/setup.py.cmakein [hunk 3]
```diff
@@ -16,8 +16,11 @@
 import setuptools
 from setuptools.command.install import install
 
-with open("../../README.md", "r") as fh:
-    long_description = fh.read()
+try:
```
Comment (commodo) [inline bindings/python/setup.py.cmakein]: this part of commit `Require python3 in CI and have fallback on readme import for setuptools ` could go into the first commit;
it's a fix commit;

**Comment by commodo** [comment]
so i went through all commits first;
it looks like these 5 commits are 2:
1. the change to `bindings/python/setup.py.cmakein`
2. the change to `.travis.yml` for CI

some descriptions would be nice to the commits;

File: bindings/python/setup.py.cmakein [hunk 4]
```diff
@@ -13,11 +13,77 @@
...
+        cmdclass={"install": InstallWrapper},
+        classifiers=[
+            "Programming Language :: Python :: 3",
+            "License :: OSI Approved :: GNU Lesser General Public License v2 or later (LGPLv2+)",
+            "Operating System :: OS Independent",
+        ],
+    )
+)
+
+
```
Comment (commodo) [inline bindings/python/setup.py.cmakein]: nitpick: extra line

File: bindings/python/setup.py.cmakein [hunk 5]
```diff
@@ -13,11 +13,77 @@
...
+        if "Windows" in _system():
+            _iiolib = "libiio.dll"
+        else:
+            # Non-windows, possibly Posix system
+            _iiolib = "iio"
+        try:
+            _lib = _cdll(find_library(_iiolib), use_errno=True, use_last_error=True)
+            if not _lib._name:
+                raise
+        except:
```
Comment (commodo) [inline bindings/python/setup.py.cmakein]: curios: it looks like this is a catch-all-exception handler; 
any idea if the `CDLL` could through other exception types that would be worth printing?
maybe it makes debugging easier when looking in CI logs later;;

**Comment by commodo** [comment]
rest looks good

**Comment by tfcollins** [comment]
ah nuts. git barfed a bit on the rebase. I'll fix this in the morning when I get in

**Comment by tfcollins** [comment]
Moving to separate PR https://github.com/analogdevicesinc/libiio/pull/380

**Pull request closed without merging**